packages/Chem/detectors.js

```diff
@@ -18,21 +21,47 @@ class ChemPackageDetectors extends DG.Package {
   detectRDSmiles(col) {
     if (col.type !== DG.TYPE.STRING)
       return null;
-
-    let name = col.name.toLowerCase();
-    if (ChemPackageDetectors.likelyNames.some((likelyName) => name.endsWith(likelyName))) {
-      col.semType = DG.SEMTYPE.MOLECULE;
-
-      // smiles or molblock?
-      let str = col.length > 0 ? col.get(0) : null;
-      col.tags[DG.TAGS.UNITS] = (str !== null && str.includes('M  END')) ? DG.UNITS.Molecule.MOLBLOCK : DG.UNITS.Molecule.SMILES;
-
-      return col.semType;
+    // return DG.SEMTYPE.MOLECULE;
+    for (let i = 0; i < col.length; ++i) {
+      const el = col.get(i);
+      if (el && el.length > 0) {
+        if (!ChemPackageDetectors.validFirstChar(el[0]))
```

Comment:
I have a feeling it would be a lot more efficient to use charCodeAt() instead of creating a one-character string

Reply:
it uses charCodeAt() inside validFirstChar() function